sdcm/utils/aws_utils.py

```diff
@@ -283,15 +283,71 @@ def init_db_info_from_params(db_info: dict, params: dict, regions: List, root_de
     return db_info
 
 
+class EC2NetworkConfiguration:
+    def __init__(self, regions: list[str], availability_zones: list[str], params: dict):
+        self.regions = regions
+        self.availability_zones = availability_zones
+        self.params = params
+        self.network_interfaces_count = network_interfaces_count(params)
+
+    @cached_property
+    def subnets_per_region(self) -> dict:
+        ec2_subnet_ids = {}
+        for region in self.regions:
+            aws_region = AwsRegion(region_name=region)
+            ec2_subnet_ids[region] = {}
+            for availability_zone in self.availability_zones:
+                ec2_subnet_ids[region][availability_zone] = self.subnets_per_availability_zone(region=region,
+                                                                                               aws_region=aws_region,
+                                                                                               availability_zone=availability_zone)
+        return ec2_subnet_ids
+
+    @cached_property
```

Comment:
there no real meaning for the caches in this class, i.e. this chace is per instance 
and the instance isn't really reused, each single property here is called once

and the lifespan of EC2NetworkConfiguration instance is quite short lived...

so I think using regault properties would be good enough

Reply:
changed